planner/core/rule_column_pruning.go

```diff
@@ -241,6 +241,17 @@ func (p *LogicalUnionAll) PruneColumns(parentUsedCols []*expression.Column) erro
 				p.schema.Columns = append(p.schema.Columns[:i], p.schema.Columns[i+1:]...)
 			}
 		}
+		// It's possible that the child operator adds extra columns to the schema.
```

Comment:
Is it possible that `children[0]` is `LogicalAggregation` and `child[1]` is `LogicalProjection` which will not add extra column?

Reply:
Switched to a new method to avoid errors in such cases.